### PR TITLE
Enable coverity scan when committing to coverity_scan

### PR DESCRIPTION
**Pull request by pmatos**

### CODE REVIEW

**Comment by pmatos** [comment]
@samth I might need a hand ensure the setup is correct as I have no commit permissions on racket. This only works when we push to a `coverity_scan` branch. Can you try doing?
```
git checkout wip-coverity-scan
git checkout -b coverity_scan
git push
```

This should trigger the analysis if all goes well. We can change the name of the *special* branch if you think `coverity_scan` is not appropriate.

File: .travis.yml [hunk 1]
```diff
@@ -58,3 +63,21 @@ notifications:
...
+env:
+  global:
+    secure: cSKI7GYTFP8VQZYMT6Fo0XQjx3r+yXf0b1r3y8NNWXOXg8xIAXFE3lM5+kvv/eDPz43sknskwmb0e093WzKN6Ht9ngJ1nsUh/cleFwPjGgYiYF4/EfMR5MhNrWQyYSbT0g2gF9+jJG1mOmXZ3cvMVzkuuJluOPB1E8tZX9T+ml8=
+
+addons:
+  coverity_scan:
+    project:
+      name: racket/racket
+      description: The Racket Programming Language
+    notification_email: pmatos@linki.tools
```
Comment (avelino) [inline .travis.yml]: Your personal e-mail?

File: .travis.yml [hunk 2]
```diff
@@ -58,3 +63,21 @@ notifications:
     secure: A19kphrabQHO8TU6qZcBaLQxdSNpm1ypEtbQsh8Ucg6HYPP7y1q7O7JZEndoMRHE9CNKZ9oXQzqR8H1IFVTlnjFFIJfkZzZ1YSNk4abSomhpWCq9daKMfwlcuTtY6PeI1nDVpka4/hiJGn9qzmaKYXle9Sl4CX2VEYp8o8PgMEs=
     rooms:
       secure: FsKzp4ItmOqd/YxqgsElgfjGW2/TU03p2p3ss+PQl/pKDQNnR/2b4pWCQ7GuqYibkmtiH1jYwrnuaLN4Cc+JyN7Z+zUtO4VSORsh3zt/gTsfgphMpCP6cB4sTqUh6AWsZOgzikj+fh7ORHEXVswQwlRHErTgZVEdEkWHBh4UWzc=
+
+before_install:
+- test $TRAVIS_BRANCH != coverity_scan -o ${TRAVIS_JOB_NUMBER##*.} = 1 || exit 0
+- echo -n | openssl s_client -connect https://scan.coverity.com:443 | sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' | sudo tee -a /etc/ssl/certs/ca-
```
Comment (avelino) [inline .travis.yml]: https://scan.coverity.com:443
No need port declared

File: .travis.yml (hunk 1 above)
Comment (pmatos) [reply .travis.yml]: Thanks. I forgot to change that. However, now that I think about it, actually my email is probably correct as I would be heading this effort as discussed with @samth . Maybe @samth  wants to get Coverity results as well?

File: .travis.yml (hunk 2 above)
Comment (pmatos) [reply .travis.yml]: I copy/pasted the suggestion from Coverity and saw no need to change that.

**Comment by samth** [comment]
I've pushed the branch, and I'm fine with you getting the results. One question -- is it possible to do this without having to regularly push to a different branch?

**Comment by pmatos** [comment]
So, the thing static analysis is very expensive and therefore there are quotas. For a project this size we are talking about two checks per day - 14 per week. Therefore the way I thought about doing it is, we have this `dummy` repo for the checks and if this build goes through, I add to this PR a script to regularly merge master to coverity_scan - once a day for example to trigger the build. However, if you have any other suggestion, I am all ears.

**Comment by pmatos** [comment]
So, the thing static analysis is very expensive and therefore there are quotas. For a project this size we are talking about two checks per day - 14 per week. Therefore the way I thought about doing it is, we have this `dummy` repo for the checks and if this build goes through, I add to this PR a script to regularly merge master to coverity_scan - once a day for example to trigger the build. However, if you have any other suggestion, I am all ears.

**Comment by samth** [comment]
Sure, I understand that we don't want to run it on every commit. But is there a way to configure that on the Coverity end so that we don't have to run something every day?

**Comment by pmatos** [comment]
@samth can you please merge my latest patch into `coverity_scan`?

**Comment by samth** [comment]
Note that you can create a PR for that branch, which makes it easier for me to merge.

**Comment by pmatos** [comment]
@samth of course, makes sense.

**Comment by jackfirth** [comment]
@pmatos I think travis has a way to run nightly CI tasks, regardless of how many (if any) commits were pushed during the day. Would that work as a way to automate the coverity integration?

**Comment by pmatos** [comment]
@jackfirth Yes, that would work!

**Comment by pmatos** [comment]
I am waiting for some information from Synopsys with regards to connecting the dashboard of coverity to github, which is currently not working. That's, I think, also causing issues with the build.

**Comment by pmatos** [comment]
Travis is failing on the `coverity_scan` branch with:
```
Coverity Scan analysis selected for branch coverity_scan.
Coverity Scan API access denied. Check $PROJECT_NAME and $COVERITY_SCAN_TOKEN.
```

This is exactly the problem I am trying to sort out with Coverity.

**Comment by pmatos** [comment]
I wanted to provide a quick update on this. I have been tracking upstream on LinkiTools/racket repo, but the coverity scan stopped working. Initially I stopped being able to access the defects dashboard and when I tried to rerun the analysis I got the email:

> Subject: Coverity Scan: Analysis failed for LinkiTools/racket
> 
>  Your request for analysis of LinkiTools/racket  is failed.
>     Analysis status: No action need
> 
>     The problem is on the Coverity server, and we will address it.
>     Your request will be automatically submitted for analysis.
>     If you do not see your build being analyzed, please email scan-admin@coverity.com.

I have tried to contact them several times for the past 2 weeks without getting any reply. So coverity scanning is a bit stuck at the moment.

**Comment by pmatos** [comment]
This is now done using an open source tool on gitlab. Lets close.

**Pull request closed without merging**